lib/oauth/error-responder.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var _ = require('lodash');
+var url = require('url');
+
+var helpers = require('../helpers');
+var forms = require('../forms');
+var resolveStateToken = require('./common').resolveStateToken;
+
+/**
+ * oauthErrorResponder takes an error object and respondes
+ * either by rendering the login form with the error, or
+ * by returning the error as JSON.
+ *
+ * @method
+ *
+ * @param {Object} req - The http request.
+ * @param {Object} res - The http response.
+ * @param {Object} err - The error to handle.
+ */
+function oauthErrorResponder(req, res, err) {
+  var accepts = req.accepts(['html', 'json']);
+  var config = req.app.get('stormpathConfig');
+
+  function renderLoginFormWithError(err) {
+    var view = req.app.get('stormpathConfig').web.login.view;
+    var nextUri = url.parse(req.query.next || '').path;
+    var formActionUri = config.web.login.uri + (nextUri ? ('?next=' + nextUri) : '');
+    var oauthStateToken = resolveStateToken(req, res);
+
+    var hasSocialProviders = _.some(config.socialProviders, function (socialProvider) {
+      return socialProvider.enabled;
+    });
+
+    if (err.code === 7201) {
+      // Stormpath is unable to create or update the account because the
```

Comment:
I'd place this comment above `if (err.code === 7201) {`. Since it's related to that line.